### DISCUSSION TITLE
Compatibility Issue with Symfony\Console v5 in Laravel 11

### DISCUSSION THREAD

**Discussion by F-Yousri** [deployphp/deployer#3926] [Bugs] [answered]
### Deployer Version

v7.5

### Target OS

macOs Sequoia 15.0.1

### Which PHP version are you using?

PHP 8.3

### Description
I encountered an issue when installing the latest version of the deployer package in a Laravel 11 application. The new package version requires `symfony/console` v5, which leads to an error during the `composer dump-autoload` process.

### Error Details
The error stems from a method signature mismatch in the `Illuminate\Console\OutputStyle` class:

```plaintext
Illuminate\Console\OutputStyle::writeln(Traversable|array|string $messages, int $type = self::OUTPUT_NORMAL): void
```

This method must be compatible with the following method from Symfony:

```plaintext
Symfony\Component\Console\Style\SymfonyStyle::writeln($messages, int $type = self::OUTPUT_NORMAL)
```

### Proposed Solution
Upgrading to `symfony/console` v7.1 would resolve this issue, as it aligns the method signatures between the two packages.

Thank you for your attention to this matter!

### Steps to reproduce

1. Create a new Laravel 11 application.
2. Install the latest version of the deployer package.
3. Run `composer dump-autoload`.

**Comment by antonmedv** [comment, marked as answer]
Rignt now you can use deployer.phar as updates for Dependencies will be only in next Deployer v8.
